online-shop/src/app/modules/shopping-cart/components/presentational/shopping-cart-details-view/shopping-cart-details-view.component.ts

```diff
@@ -0,0 +1,30 @@
+import { Component, Input, OnInit } from '@angular/core';
+import { Product, OrderProduct } from 'src/app/modules/shared/types/products.types';
+
+@Component({
+  selector: 'app-shopping-cart-details-view',
+  templateUrl: './shopping-cart-details-view.component.html',
+  styleUrls: ['./shopping-cart-details-view.component.scss']
+})
+export class ShoppingCartDetailsViewComponent implements OnInit {
+  @Input() products!: Product[]
+  orderProducts: OrderProduct[] = []
+
+  ngOnInit() {
+    this.orderProducts = this.products.map((product) => ({
+      product,
+      quantity: 1,
+    }));
+  }
+
+  onRemoveButtonClick(orderProduct: OrderProduct) {
+    const index = this.orderProducts.indexOf(orderProduct);
```

Comment:
Your version is also correct, but I always had a had time in reading the logic of "-1". Do not seems that intuitive to me, but just a personal opinion :) I prefer something like this:

```
if (this.orderProducts.includes(orderProduct)) {
  this.orderProducts.splice(this.orderProducts.indexOf(orderProduct), 1);
}
```

Reply:
I understand, i changed it